net-gateway/src/endpoints/files/pcap_files.rs

```diff
@@ -67,13 +72,54 @@ async fn pcap_files(
         let request = Envelope::new(
             tenant_id,
             packet_data.get_type(),
-            &packet_data.encode());
+            &packet_data.encode()
+        );
         
         match server_connection.send_all_reliable(&request.encode()).await {
             Ok(_) => (),
             Err(e) => return HttpResponse::InternalServerError().body(e.to_string()),
         };
+
+        let response = match server_connection.receive_reliable().await {
+            Ok(response) => ResultDTO::decode(&response),
+            Err(err) => return HttpResponse::InternalServerError().json(serde_json::json!({
+                "error": "Invernal Server Error",
+                "message": err.to_string()
+            })),
+        };
+        match response.is_ok() {
+            true => {
+                let response = response.into_inner();
+                if response.is_none() {
+                    return HttpResponse::InternalServerError().json(serde_json::json!({
+                        "error": "Invernal Server Error",
+                        "message": "Coldn't get network packet",
+                    }));
+                }
+                let response = response.unwrap();
+                if response.get_envelope_type() != NetworkPacketDTO::get_data_type() {
+                    return HttpResponse::InternalServerError().json(serde_json::json!({
+                        "error": "Invernal Server Error",
+                        "message": "Received wrong data type after decoding, double check the data you want to decode"
+                    }));
+                }
+                packets.push(NetworkPacketDTO::decode(response.get_data()).into());
```

Comment:
Dont you wanna make this one work in parallel?

Reply:
Like, I mean...
Youll prob understand me

Reply:
Do you mean decoding packets?